src/index.jsx

```diff
@@ -1,95 +1,62 @@
-import {
-  identifyAuthenticatedUser,
-  sendPageEvent,
-  configureAnalytics,
-  initializeSegment,
-} from '@edx/frontend-analytics';
-import { configureLoggingService, NewRelicLoggingService } from '@edx/frontend-logging';
-import { getAuthenticatedAPIClient } from '@edx/frontend-auth';
-import { configure as configureI18n } from '@edx/frontend-i18n';
+import { Route, Switch } from 'react-router-dom';
 
-import { configuration } from './environment';
-import messages from './i18n';
-import configureStore from './store';
-import { configureUserAccountApiService } from './common';
-import getQueryParameters from './data/getQueryParameters';
-import { configureApiService as configurePaymentApiService } from './payment';
+import Header, { messages as headerMessages } from '@edx/frontend-component-header';
+import Footer, { messages as footerMessages } from '@edx/frontend-component-footer';
 
+import appMessages from './i18n';
+import { PaymentPage, EcommerceRedirect, responseInterceptor } from './payment';
+import configureStore from './store';
 import './index.scss';
-import App from './components/App';
-import ErrorPage from './common/components/ErrorPage';
-
-let apiClient = null;
-
-/**
- * We need to merge the application configuration with the authentication state
- * so that we can hand it all to the redux store's initializer.
- */
-function createInitialState(authenticatedUser) {
-  const queryParameters = getQueryParameters();
-  const authenicationState = { authentication: authenticatedUser };
-  return Object.assign({}, { configuration, queryParameters }, authenicationState);
-}
-
-function configure(authenticatedUser) {
-  configureI18n(configuration, messages);
-
-  const { store, history } = configureStore(createInitialState(authenticatedUser), configuration.ENVIRONMENT);
 
-  configureLoggingService(NewRelicLoggingService);
-  configurePaymentApiService(configuration, apiClient);
-  configureUserAccountApiService(configuration, apiClient);
-  initializeSegment(configuration.SEGMENT_KEY);
-  configureAnalytics({
-    loggingService: NewRelicLoggingService,
-    authApiClient: apiClient,
-    analyticsApiBaseUrl: configuration.LMS_BASE_URL,
-  });
-
-  return {
-    store,
-    history,
-  };
-}
-
-function initialize(authenticatedUser) {
-  const { store, history } = configure(authenticatedUser);
-
-  ReactDOM.render(<App store={store} history={history} />, document.getElementById('root'));
-  identifyAuthenticatedUser(authenticatedUser.userId);
-  sendPageEvent();
-}
-
-try {
-  if (configuration.ENVIRONMENT !== 'production') {
-    import(/* webpackChunkName: "react-axe" */ 'react-axe').then(({ default: axe }) =>
-      axe(React, ReactDOM, 1000));
+App.subscribe(APP_READY, () => {
+  ReactDOM.render(
+    <AppProvider store={configureStore()}>
+      <Header />
+      <main>
+        <Switch>
+          <Route exact path="/" component={PaymentPage} />
+          <Route path="*" component={EcommerceRedirect} />
```

Comment:
Could we use `<Redirect />` from react-router-dom?

https://reacttraining.com/react-router/web/api/Redirect

Reply:
`EcommerceRedirect` also logs... that said, it could internally use `Redirect`.  

I think I may leave it, just to minimize potential changes to the actual functionality.